lib/junit.rb

```diff
@@ -0,0 +1,71 @@
+# frozen_string_literal: true
+
+module AutoHCK
+  class JUnit
+    template = Erubi::Engine.new(File.read('lib/templates/junit.xml.erb'), escape: true)
+
+    # rubocop:disable Security/Eval
+    eval <<-RUBY, binding, __FILE__, __LINE__ + 1
+      # def build_xml
+      #   Erubi template compiled source code
+      # end
+
+      def build_xml
+        #{template.src}
+      end
+    RUBY
```

Comment:
I guess you intended to make a stack trace generated in the template point to line 10, but it will point to a line after line 14.

A line number of an Erubi's output apparently matches the template source code, so I suggest the following:
```suggestion
    path = 'lib/templates/junit.xml.erb'
    binding.eval "def build_xml; #{template.src}; end", path
```

By the way, looking at Erubi's output, I came up with a trivial suggestion for the template so I leave the comments for it too.

Reply:
Oneline is bad due to rubocop offence Style/DocumentDynamicEvalDefinition, this is the reason for the comment.

Reply:
```
    template = Erubi::Engine.new(File.read('lib/templates/junit.xml.erb'), escape: true)
    path = 'lib/templates/junit.xml.erb'

    # rubocop:disable Security/Eval
    binding.eval <<-RUBY, path
      # def build_xml
      #   Erubi template compiled source code
      # end

      def build_xml
        #{template.src}
      end
    RUBY
    # rubocop:enable Security/Eval
```

Reply:
The multi-line heredoc breaks the line number. Considering the code is so trivial, I think it's OK to disable Style/DocumentDynamicEvalDefinition here.

Reply:
Fixed